genie-proto/src/main/proto/genie.proto

```diff
@@ -145,6 +154,56 @@ message JobSpecificationServiceError {
 message JobSpecificationResponse {
     oneof response {
         JobSpecification specification = 1;
-        JobSpecificationServiceError error = 2;
+        JobSpecificationError error = 2;
+    }
+}
+
+message AgentMetadata {
+    string agent_version = 1;
+    string agent_hostname = 2;
+    string agent_pid = 3;
+}
+
+message ClaimJobRequest {
+    string id = 1;
+    AgentMetadata agent_metadata = 2;
+}
+
+message ClaimJobError {
+    enum Type {
+        UNKNOWN = 0;
+        ALREADY_CLAIMED = 1;
+        NO_SUCH_JOB = 2;
+    }
+    Type type = 1;
+    string message = 2;
+}
+
+message ClaimJobResponse {
+    oneof response {
+        bool successful = 1;
+        ClaimJobError error = 2;
+    }
+}
+
+message ChangeJobStatusRequest {
+    string current_status = 1;
```

Comment:
This could be a protobuf enum clone of the JobStatus enum, ugly to duplicate, but more type safe and explicit, wdyt?

Reply:
I avoided doing this for the feeling of copying and pasting over and over. We can do it for type safety but I was thinking since it's internal API we can still do something like `JobStatus.RUNNING.toString()` on one side and `JobStatus.parse(string)` on the other side to ensure the types are consistent. This is just a wire protocol.